doc/radosgw/cloud-transition.rst

```diff
@@ -0,0 +1,331 @@
+user id to a destination id.
+
+* ``type`` (id | email | uri)
+
+ACL type: ``id`` defines user id, ``email`` defines user by email, and ``uri`` defines user by ``uri`` (group).
+
+* ``source_id`` (string)
+
+ID of user in the source zone.
+
+* ``dest_id`` (string)
+
+ID of user in the destination.
+
+* ``target_path`` (string)
+
+A string that defines how the target path is created. The target path specifies a prefix to which
+the source 'bucket-name/object-name' is appended. If not specified the target_path created is "rgwx-${zonegroup}-${storage-class}-cloud-bucket".
+
+For example: ``target_path = rgwx-archive-${zonegroup}/``
+
+* ``target_storage_class`` (string)
+
+A string that defines the target storage class to which the object transitions to. If not specified, object is transitioned to STANDARD storage class.
+
+* ``retain_object`` (true | false)
+
+If true, retains the metadata of the object transitioned to cloud. If false (default), the object is deleted post transition.
+
+
+S3 Specific Configurables:
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently cloud transition will only work with backends that are compatible with AWS S3. There are
+a few configurables that can be used to tweak its behavior when accessing these cloud services:
+
+::
+
+    {
+      "multipart_sync_threshold": {object_size},
+      "multipart_min_part_size": {part_size}
+    }
+
+
+* ``multipart_sync_threshold`` (integer)
+
+Objects this size or larger will be transitioned to the cloud using multipart upload.
+
+* ``multipart_min_part_size`` (integer)
+
+Minimum parts size to use when transitioning objects using multipart upload.
+
+
+How to Configure
+~~~~~~~~~~~~~~~~
+
+See `Adding a Storage Class <https://docs.ceph.com/en/latest/radosgw/placement/#adding-a-storage-class>`__ for how to configure storage-class for a zonegroup. The cloud transition requires a creation of a special storage class with tier type defined as ``cloud-s3``
+
+Note: Once a storage class is created of ``--tier-type=cloud-s3``, it cannot be later modified to any other storage class type.
+
+::
+
+    # radosgw-admin zonegroup placement add --rgw-zonegroup={zone-group-name} \
+                                            --placement-id={placement-id} \
+                                            --storage-class={storage-class-name} \
+                                            --tier-type=cloud-s3 
+
+For example:
+
+::
+
+    # radosgw-admin zonegroup placement add --rgw-zonegroup=default \
+                                            --placement-id=default-placement \
+                                            --storage-class=CLOUDTIER --tier-type=cloud-s3
+    [
+        {
+            "key": "default-placement",
+            "val": {
+                "name": "default-placement",
+                "tags": [],
+                "storage_classes": [
+                    "CLOUDTIER",
+                    "STANDARD"
+                ],
+                "tier_targets": [
```

Comment:
@cbodley @yehudasa  ... As per your comment in the last refactoring call, changed the supported tier-type to 'cloud-s3'. Below are the options available for this tier-type, which can be configured as <key,value> pair (line: 175).  Kindly review.